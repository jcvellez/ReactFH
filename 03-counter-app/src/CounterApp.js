import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ numero=10 }) => {

    let [COUNTER, setCOUNTER] = useState(numero);
   
    let handleAdd = () => {          
        setCOUNTER(COUNTER+1);
        //setCOUNTER((c)=>c+1);
        //COUNTER+1     
    };
    let handleSubstract = () => {          
        setCOUNTER(COUNTER-1);
        //setCOUNTER((c)=>c+1);
        //COUNTER+1     
    };
    let handleReset = () => {          
        setCOUNTER(COUNTER=numero);
        //setCOUNTER((c)=>c+1);
        //COUNTER+1     
    };


    return (
        <>
            <h1>CounterApp</h1>
            <h2 id="contador">{COUNTER}</h2>
            <button id="botonADD" onClick={handleAdd}>+1</button>
            <button id="botonRESET" onClick={handleReset}>Reset</button>
            <button id="botonSUB" onClick={handleSubstract}>-1</button>
        </>
    );
};

CounterApp.propTypes = {
    numero: PropTypes.number
}

export default CounterApp;