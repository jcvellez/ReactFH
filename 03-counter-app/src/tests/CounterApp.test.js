import React from 'react'
import { render } from "@testing-library/react";
import CounterApp from "../CounterApp";
import { shallow } from 'enzyme';

describe('Pruebas en <CounterApp />', () => {

    let wraper;
    beforeEach(() => {
        wraper = shallow(<CounterApp />);
    })

    // test('Debo mostrar <CounterApp /> correctamente', () => {
    //     //const numero = 100;    
    //     expect(wraper).toMatchSnapshot();
    // })

    // test('Tomar el valor del counter del elemento h2', () => {
    //     const titulo = "CounterApp";
    //     const contador = 10;
    //     const wraper = shallow(
    //         <CounterApp
    //             titulo={titulo}
    //             contador={contador}
    //         />
    //     );
    //     const COUNTER = wraper.find('#contador').text();
    //     expect(parseInt(COUNTER)).toEqual(contador);
    // })

    // test('Debe incrementar el contador en 1 (+1)', () => {
    //     //const botonAdd= wraper.find('button').at(0);
    //     wraper.find('#botonADD').simulate('click');

    //     const COUNTERText = wraper.find('#contador').text().trim();
    //     expect(COUNTERText).toEqual('11');
    //     console.log('COUNTERText es ', COUNTERText);
    // })

    // test('Debe decrementar el contador en 1 (-1)', () => {
    //     wraper.find('#botonSUB').simulate('click');
    //     const COUNTERText = wraper.find('#contador').text().trim();
    //     expect(COUNTERText).toEqual('9');
    // })

    test('Debe recibir el valor de reset', () => { 
        const wraper = shallow(<CounterApp numero={108}/>);
        wraper.find('#botonADD').simulate('click');
        const COUNTERText = wraper.find('#contador').text().trim();
        console.log(COUNTERText);

        // const botonRESET= wraper.find('#botonRESET').simulate('click');
        // const COUNTERText = wraper.find('#contador').text().trim();
        // console.log(COUNTERText);
        // expect(COUNTERText).toEqual('10');
     })

})