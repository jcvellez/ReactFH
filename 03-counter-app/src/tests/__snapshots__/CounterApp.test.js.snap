// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debo mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="contador"
  >
    10
  </h2>
  <button
    id="botonADD"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="botonRESET"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="botonSUB"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
